mars-rover-new/src/mars_rover_new/core.clj

```diff
@@ -90,6 +112,28 @@
   {:pre [(valid-cmd-sequence? cmd-sequence)]}
   (assoc mars-rover-data :cmd-sequence cmd-sequence))
 
+(defn rotate-left
+  "Rotate rover counter-clockwise 90 degrees"
+  [mars-rover-data]
+  (assoc-in mars-rover-data [:rover-position :direction] (get-in transforms [:left (get-in mars-rover-data [:rover-position :direction])])))
+
+(defn rotate-right
+  "Rotate rover clockwise 90 degrees"
+  [mars-rover-data]
+  (assoc-in mars-rover-data [:rover-position :direction] (get-in transforms [:right (get-in mars-rover-data [:rover-position :direction])])))
+
+(defn move
+  "Move rover forward one unit"
+  [mars-rover-data]
+  (let [direction (get-in mars-rover-data [:rover-position :direction])
+        {key :coord op :action} (get-in transforms [:move direction])]
+    (assoc-in mars-rover-data [:rover-position key] (op (get-in mars-rover-data [:rover-position key])))))
+
+;; mapping of L, R and M to their respective core functions
+(def fn-map {"L" mars-rover-new.core/rotate-left
```

Comment:
Consider renaming `fn-map`, the comment explains it pretty well but strive to name functions which are self explanatory.